source/test/TestApp/PlayFabEventTest.cpp

```diff
@@ -91,12 +91,23 @@ namespace PlayFabUnit
     int PlayFabEventTest::eventPassCount;
     int PlayFabEventTest::eventFailCount;
     std::string PlayFabEventTest::eventFailLog;
+    unsigned int PlayFabEventTest::numEventsHeard;
+    std::vector<std::thread> PlayFabEventTest::testThreadPool;
 
     void PlayFabEventTest::NonStaticEmitEventCallback(std::shared_ptr<const PlayFab::IPlayFabEvent> /*event*/, std::shared_ptr<const PlayFab::IPlayFabEmitEventResponse> /*response*/)
     {
         (*eventTestContext)->Pass("Private member called back!");
     }
 
+    void PlayFabEventTest::MyThreadingCallback(std::shared_ptr<const PlayFab::IPlayFabEvent> /*event*/, std::shared_ptr<const PlayFab::IPlayFabEmitEventResponse> /*response*/)
+    {
+        numEventsHeard++;
+        if(numEventsHeard == numTotalThreadedEvents)
+        {
+            (*eventTestContext)->Pass("Threaded callback worked as expected!");
```

Comment:
>Pass [](start = 33, length = 4)

what scenarios would lead to this test ever failing (not counting halting indefinitely)?

Reply:
Only possible failure state is a time out when we didn't hit that expected limit. 

---
In reply to: [345446233](https://github.com/PlayFab/XPlatCoreTemplate/pull/44#discussion_r345446233) [](ancestors = 345446233)